src/Payload.cpp

```diff
@@ -907,6 +928,39 @@ omnmmsgPayload_unSerialize (const msgPayload    msg,
     // Move tail to end of buffer
     impl->mPayloadBufferTail = bufferLength;
 
+    impl->mFieldHints.clear();
```

Comment:
This whole preparsing sectionlooks expensive and should at least be optional. It's iterating the entire buffer before the app has even seen it and populates an STL object with an index. This will penalize well optimized OpenMAMA applications and reward poorly optimized OpenMAMA applications.

Reply:
Looks like appearances could be deceiving - this branch certainly does look faster on first batch of tests - nice.